sql/catalyst/src/main/scala/org/apache/spark/sql/connector/catalog/Catalogs.scala

```diff
@@ -39,13 +39,17 @@ private[sql] object Catalogs {
    * @param conf a SQLConf
    * @return an initialized CatalogPlugin
    * @throws CatalogNotFoundException if the plugin class cannot be found
-   * @throws org.apache.spark.SparkException           if the plugin class cannot be instantiated
+   * @throws org.apache.spark.SparkException if the plugin class cannot be instantiated
    */
   @throws[CatalogNotFoundException]
   @throws[SparkException]
   def load(name: String, conf: SQLConf): CatalogPlugin = {
     val pluginClassName = try {
-      conf.getConfString("spark.sql.catalog." + name)
```

Comment:
Can we do the validation in `CatalogManager`? This place seems too low-level.

Reply:
nvm, it's better to do it here, so that we don't need to validate the name at each catalog access (the result is cached)

Reply:
After a second thought, this name limitation is only for the current config-based way to register catalogs. Now I think your first version is better: we only need to forbid dot in the catalog name. Maybe in the future, we add new ways to register catalogs and have different limitations.

Can you change back to the first version? Sorry for the back and forth!

Reply:
Got it, reverted to forbid dot.